bin.src/make_latiss_export.py

```diff
@@ -0,0 +1,110 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+"""Selectively export the contents of the LATISS dataset.
+
+This script selects some LATISS data in a source butler repo, and makes an export
+file for importing to the test central prompt processing repository.
+"""
+
+
+import argparse
+import logging
+import sys
+import time
+
+import lsst.daf.butler as daf_butler
+
+
+def _make_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--src-repo",
+        required=True,
+        help="The location of the repository to be exported.",
+    )
+    return parser
+
+
+def main():
+    logging.basicConfig(level=logging.INFO, stream=sys.stdout)
+
+    args = _make_parser().parse_args()
+    src_butler = daf_butler.Butler(args.src_repo)
+
+    logging.info("Exporting datasets and collections")
+    start = time.time_ns()
+    _export_for_copy(src_butler)
+    end = time.time_ns()
+    logging.info("Export finished in %.3fs.", 1e-9 * (end - start))
+
+
+def _export_for_copy(butler):
+    """Export selected data to make copies in another butler repo.
+
+    Parameters
+    ----------
+    butler: `lsst.daf.butler.Butler`
+        The source Butler from which datasets are exported.
+    """
+    with butler.export(format="yaml") as contents:
+        logging.debug("Selecting deepCoadd datasets")
+        records = butler.registry.queryDatasets(
+            datasetType="deepCoadd",
+            collections="LATISS/runs/AUXTEL_DRP_IMAGING_2023-07AB-05AB/"
+            + "w_2023_19/PREOPS-3598/20230726T202836Z",
+        )
+        contents.saveDatasets(records)
+
+        logging.debug("Selecting refcats datasets")
+        records = butler.registry.queryDatasets(datasetType=..., collections="refcats")
+        contents.saveDatasets(records)
+
+        logging.debug("Selecting skymaps dataset")
+        records = butler.registry.queryDatasets(
+            datasetType="skyMap", collections="skymaps", dataId={"skymap": "latiss_v1"}
+        )
+        contents.saveDatasets(records)
+
+        logging.debug("Selecting datasets in LATISS/calib")
+        records = butler.registry.queryDatasets(
+            datasetType=..., collections="LATISS/calib"
+        )
+        contents.saveDatasets(records)
+
+        # Save collections and chains for calibration and templates
+        for collection in butler.registry.queryCollections(
+            expression="LATISS/calib",
+            flattenChains=True,
+        ) + [
+            "LATISS/calib",
+            "LATISS/calib/DM-36719",
+            "LATISS/calib/DM-38946",
+            "LATISS/calib/DM-39505",
+            "LATISS/templates",
+        ]:
```

Comment:
Should `refcats` also be in this list? That should make it unnecessary to edit the chain by hand.

Reply:
Because as-it it only saves the collection if the collection doesn't already exist (that'd mean all of these LATISS collections this time), and the `refcats` collection already exists in the dev central repo, I'd still edit the chain by hand this time instead of changing the script for it.